Consider moving @types/bn.js to dependencies?

It's currently a `devDependency`, which is perfectly fine for an application but for a library falls in a gray area. In theory it should be a `devDependency`, but since [BG is exposed](https://github.com/ethereumjs/ethereumjs-util/blob/599ba5b1c7043a7e155e6032c50d7a01fc63aaf1/src/index.ts#L10) it leads to build failures on projects that use TypeScript and depend on `ethereumjs-util`.

See [example](https://gist.github.com/lautarodragan/e165128d34e4551cc233440285dd81ae).

```
> example-ethereumjs-util-issue@1.0.0 build /home/lautarodragan/Projects/example-ethereumjs-util-issue
> tsc

node_modules/ethereumjs-util/dist/index.d.ts:1:21 - error TS7016: Could not find a declaration file for module 'bn.js'. '/home/lautarodragan/Projects/example-ethereumjs-util-issue/node_modules/bn.js/lib/bn.js' implicitly has an 'any' type.
  Try `npm install @types/bn.js` if it exists or add a new declaration (.d.ts) file containing `declare module 'bn.js';`

1 import BN = require('bn.js');
                      ~~~~~~~

node_modules/rlp/dist/types.d.ts:2:21 - error TS7016: Could not find a declaration file for module 'bn.js'. '/home/lautarodragan/Projects/example-ethereumjs-util-issue/node_modules/bn.js/lib/bn.js' implicitly has an 'any' type.
  Try `npm install @types/bn.js` if it exists or add a new declaration (.d.ts) file containing `declare module 'bn.js';`

2 import BN = require('bn.js');
                      ~~~~~~~


Found 2 errors.
```

Running `npm install -DE @types/bn.js` does fix the issue. So, maybe just documenting the install instructions is all that's needed. But it could eventually somehow lead to odd issues if different versions are wanted by the application and ethereumjs-util? Though for the current use in [index.d.ts](https://unpkg.com/ethereumjs-util@6.1.0/dist/index.d.ts) it seems unlikely.

On the other hand including BN as a dependency would make the package heavier for everyone with no benefit for non-ts users.

Thought: no change would be needed and this would automagically go away if BN had the type declarations in their own repo and package.

--- a/package.json
+++ b/package.json
@@ -88,6 +88,7 @@
   },
   "homepage": "https://github.com/ethereumjs/ethereumjs-util",
   "dependencies": {
+    "@types/bn.js": "^4.11.3",
     "bn.js": "^4.11.0",
     "create-hash": "^1.1.2",
     "ethjs-util": "0.1.6",
@@ -99,7 +100,6 @@
     "@ethereumjs/config-prettier": "^1.1.0",
     "@ethereumjs/config-tsc": "^1.1.0",
     "@ethereumjs/config-tslint": "^1.1.0",
-    "@types/bn.js": "^4.11.3",
     "@types/node": "^11.9.0",
     "babel-cli": "^6.26.0",
     "babel-preset-env": "^1.6.1",

